contract_compassion/tests/test_contract_compassion.py

```diff
@@ -0,0 +1,339 @@
+        journal_obj = self.registry('account.journal')
+        bank_journal_id = self.registry('account.journal').search(
+            self.cr, self.uid, [('code', '=', 'TBNK')])[0]
+        bank_journal = journal_obj.browse(self.cr, self.uid, bank_journal_id)
+        move_obj = self.registry('account.move')
+        move_line_obj = self.registry('account.move.line')
+        invoice = self.registry('account.invoice').browse(
+            self.cr, self.uid, invoice_id)
+        account_id = invoice.partner_id.property_account_receivable.id
+        move_id = move_obj.create(self.cr, self.uid, {
+            'journal_id': bank_journal_id
+        })
+        move_line_obj.create(self.cr, self.uid, {
+            'name': 'BNK-' + invoice.number,
+            'move_id': move_id,
+            'partner_id': invoice.partner_id.id,
+            'account_id': bank_journal.default_debit_account_id.id,
+            'debit': invoice.amount_total,
+            'journal_id': bank_journal_id,
+            'period_id': invoice.period_id.id,
+            'date': invoice.date_due
+        })
+        mv_line_id = move_line_obj.create(self.cr, self.uid, {
+            'name': 'PAY-' + invoice.number,
+            'move_id': move_id,
+            'partner_id': invoice.partner_id.id,
+            'account_id': account_id,
+            'credit': invoice.amount_total,
+            'journal_id': invoice.journal_id.id,
+            'period_id': invoice.period_id.id,
+            'date': invoice.date_due
+        })
+        move_obj.button_validate(self.cr, self.uid, [move_id])
+        to_reconcile = move_line_obj.search(self.cr, self.uid, [
+            ('move_id', '=', invoice.move_id.id),
+            ('account_id', '=', account_id)]) + [mv_line_id]
+        move_line_obj.reconcile(self.cr, self.uid, to_reconcile)
+
+    def test_contract_compassion_first_scenario(self):
+        """
+            In this test we are testing states changement of a contract and if
+            the old invoice are well cancelled when we pay one invoice.
+        """
+        # Testing area creation
+        self.contract_group = self._create_group(
+            'do_nothing', 1, 'month', self.partner_id, 5, self.payment_term_id)
+        self.contract_id = self._create_contract(
+            datetime.today().strftime(DF), self.contract_group,
+            'phone', datetime.today().strftime(DF))
+        self.contract_line_id = self._create_contract_line(
+            self.contract_id, '2', '40.0')
+
+        contract_obj = self.registry('recurring.contract')
+        contract = contract_obj.browse(self.cr, self.uid, self.contract_id)
+        self.assertEqual(contract.state, 'draft')
+
+        # Switching to "waiting for payment" state
+        wf_service = netsvc.LocalService('workflow')
+        wf_service.trg_validate(
+            self.uid, 'recurring.contract',
+            self.contract_id, 'contract_validated', self.cr)
+        contract_wait = contract_obj.browse(

```

Comment:
Don't create a new variable for holding the same contract as before. Just use `contract` again.